### ISSUE TITLE
set_epoch on DataLoader is needed for distributed training

### ISSUE THREAD

**Issue by npuichigo** [kan-bayashi/ParallelWaveGAN#177]
[DistributedSampler](https://pytorch.org/docs/master/data.html?highlight=distributedsampler#torch.utils.data.distributed.DistributedSampler) in pytorch says that, in distributed mode, calling the `set_epoch(epoch)` method at the beginning of each epoch before creating the DataLoader iterator is necessary to make shuffling work properly across multiple epochs. Otherwise, the same ordering will be always used.

It seems that it's missing here.

Example:
```python
>>> sampler = DistributedSampler(dataset) if is_distributed else None
>>> loader = DataLoader(dataset, shuffle=(sampler is None),
...                     sampler=sampler)
>>> for epoch in range(start_epoch, n_epochs):
...     if is_distributed:
...         sampler.set_epoch(epoch)
...     train(loader)
```

**Comment by kan-bayashi** [comment]
@npuichigo Thank you for pointing out.
I will check it.
